open-apparel-registry/open-apparel-registry
# Use HubSpot for mailing list

## Overview

Replaces the existing Mailchimp integration with HubSpot.

Connects #1944 

## Demo

Dev account
<img width="828" alt="Screen Shot 2022-09-26 at 9 23 35 AM" src="https://user-images.githubusercontent.com/21046714/192287982-4c27e27e-e899-45c6-a9f8-20515822b6f9.png">

Real OSHub subscription
<img width="1001" alt="Screen Shot 2022-09-26 at 9 58 51 AM" src="https://user-images.githubusercontent.com/21046714/192296081-9950a24d-25a2-4db0-be3a-7d416af9c0ac.png">

## Notes

I added the HubSpot variables to terraform on AWS, but didn't remove Mailchimp. We'll need to remove the Mailchimp variables once this is successfully merged.

## Testing Instructions

* Run `./scripts/server`. Sign up for an account and check the mailing list checkbox. Since you have no `HUBSPOT_API_KEY` set up, you'll be registered but not added to a mailing list. There shouldn't be an error.
* I will send you the test account API key and subscription ID directly. (Note: This key is only valid for 90 days, and applies to a developer account I set up just for testing this.) Add them to your `.env` file:
```
HUBSPOT_API_KEY={key}
HUBSPOT_SUBSCRIPTION_ID={id}
```
* Sign up for an account and check the mailing list checkbox. 
* Using Postman or similar, send a GET request to `https://api.hubapi.com/communication-preferences/v3/status/email/{EMAIL}` with EMAIL being the email you used to sign up. For authentication, use the API_KEY as a Bearer token. You should receive an object with two subscriptions listed. The one for "Open Supply Hub News and Updates" should be SUBSCRIBED. 
* If desired, you can also attempt this with the real OSHub API key and subscription id (see the OSHub terraform variables), but you'll be signed up for the real email subscription with whichever email you use. 

## Checklist
